About  libcurl.a is link against libnghttp2.dll.a

it's been there for quite a long time, please fix it.

#5440
#4404
#3538
#950

After #8773 merged, last two problems fixed:
> libidn2.a should link against libunistring.a not libunistring.dll.a
> libpsl.a should link against libunistring.a not libunistring.dll.a

First one still broken:
> libcurl.a should link against libnghttp2.a not libnghttp2.dll.a

With "-lnghttp2":
```
ld: libcurl.a(libcurl_la-http2.o):(.text+0xe9): undefined reference to `__imp_nghttp2_session_want_write'
ld: libcurl.a(libcurl_la-http2.o):(.text+0x111): undefined reference to `__imp_nghttp2_session_del'
ld: libcurl.a(libcurl_la-http2.o):(.text+0x149): undefined reference to `__imp_nghttp2_session_get_stream_user_data'
ld: libcurl.a(libcurl_la-http2.o):(.text+0x18b): undefined reference to `__imp_nghttp2_session_get_stream_user_data'
ld: libcurl.a(libcurl_la-http2.o):(.text+0x4ab): undefined reference to `__imp_nghttp2_submit_rst_stream'
ld: libcurl.a(libcurl_la-http2.o):(.text+0x5b9): undefined reference to `__imp_nghttp2_session_get_stream_user_data'
ld: libcurl.a(libcurl_la-http2.o):(.text+0x6c9): undefined reference to `__imp_nghttp2_session_get_stream_user_data'
ld: libcurl.a(libcurl_la-http2.o):(.text+0x832): undefined reference to `__imp_nghttp2_session_get_stream_user_data'
ld: libcurl.a(libcurl_la-http2.o):(.text+0x881): undefined reference to `__imp_nghttp2_session_set_stream_user_data'
ld: libcurl.a(libcurl_la-http2.o):(.text+0x95a): undefined reference to `__imp_nghttp2_priority_spec_init'
ld: libcurl.a(libcurl_la-http2.o):(.text+0x9a9): undefined reference to `__imp_nghttp2_submit_priority'
ld: libcurl.a(libcurl_la-http2.o):(.text+0x9c0): undefined reference to `__imp_nghttp2_session_send'
ld: libcurl.a(libcurl_la-http2.o):(.text+0xa28): undefined reference to `__imp_nghttp2_session_mem_recv'
ld: libcurl.a(libcurl_la-http2.o):(.text+0xa61): undefined reference to `__imp_nghttp2_session_check_request_allowed'
ld: libcurl.a(libcurl_la-http2.o):(.text+0xaa9): undefined reference to `__imp_nghttp2_session_want_read'
ld: libcurl.a(libcurl_la-http2.o):(.text+0xab6): undefined reference to `__imp_nghttp2_session_want_write'
ld: libcurl.a(libcurl_la-http2.o):(.text+0xb3b): undefined reference to `__imp_nghttp2_strerror'
ld: libcurl.a(libcurl_la-http2.o):(.text+0xcbe): undefined reference to `__imp_nghttp2_submit_ping'
ld: libcurl.a(libcurl_la-http2.o):(.text+0xcdf): undefined reference to `__imp_nghttp2_session_send'
ld: libcurl.a(libcurl_la-http2.o):(.text+0xcf1): undefined reference to `__imp_nghttp2_strerror'
ld: libcurl.a(libcurl_la-http2.o):(.text+0xd14): undefined reference to `__imp_nghttp2_strerror'
ld: libcurl.a(libcurl_la-http2.o):(.text+0xed1): undefined reference to `__imp_nghttp2_http2_strerror'
ld: libcurl.a(libcurl_la-http2.o):(.text+0x10a4): undefined reference to `__imp_nghttp2_session_want_read'
ld: libcurl.a(libcurl_la-http2.o):(.text+0x10b5): undefined reference to `__imp_nghttp2_session_want_write'
ld: libcurl.a(libcurl_la-http2.o):(.text+0x13de): undefined reference to `__imp_nghttp2_session_resume_data'
ld: libcurl.a(libcurl_la-http2.o):(.text+0x13fa): undefined reference to `__imp_nghttp2_is_fatal'
ld: libcurl.a(libcurl_la-http2.o):(.text+0x18e2): undefined reference to `__imp_nghttp2_priority_spec_init'
ld: libcurl.a(libcurl_la-http2.o):(.text+0x198d): undefined reference to `__imp_nghttp2_submit_request'
ld: libcurl.a(libcurl_la-http2.o):(.text+0x1a06): undefined reference to `__imp_nghttp2_session_resume_data'
ld: libcurl.a(libcurl_la-http2.o):(.text+0x1bf0): undefined reference to `__imp_nghttp2_session_want_read'
ld: libcurl.a(libcurl_la-http2.o):(.text+0x1c01): undefined reference to `__imp_nghttp2_session_want_write'
ld: libcurl.a(libcurl_la-http2.o):(.text+0x1c94): undefined reference to `__imp_nghttp2_session_want_read'
ld: libcurl.a(libcurl_la-http2.o):(.text+0x1ca5): undefined reference to `__imp_nghttp2_session_want_write'
ld: libcurl.a(libcurl_la-http2.o):(.text+0x1dc2): undefined reference to `__imp_nghttp2_version'
ld: libcurl.a(libcurl_la-http2.o):(.text+0x1f5d): undefined reference to `__imp_nghttp2_session_get_stream_user_data'
ld: libcurl.a(libcurl_la-http2.o):(.text+0x1fa6): undefined reference to `__imp_nghttp2_submit_rst_stream'
ld: libcurl.a(libcurl_la-http2.o):(.text+0x1fae): undefined reference to `__imp_nghttp2_is_fatal'
ld: libcurl.a(libcurl_la-http2.o):(.text+0x1fd6): undefined reference to `__imp_nghttp2_session_get_remote_settings'
ld: libcurl.a(libcurl_la-http2.o):(.text+0x23d3): undefined reference to `__imp_nghttp2_submit_rst_stream'
ld: libcurl.a(libcurl_la-http2.o):(.text+0x23db): undefined reference to `__imp_nghttp2_is_fatal'
ld: libcurl.a(libcurl_la-http2.o):(.text+0x248f): undefined reference to `__imp_nghttp2_session_set_stream_user_data'
ld: libcurl.a(libcurl_la-http2.o):(.text+0x260c): undefined reference to `__imp_nghttp2_session_set_stream_user_data'
ld: libcurl.a(libcurl_la-http2.o):(.text+0x264c): undefined reference to `__imp_nghttp2_submit_rst_stream'
ld: libcurl.a(libcurl_la-http2.o):(.text+0x26b4): undefined reference to `__imp_nghttp2_session_send'
ld: libcurl.a(libcurl_la-http2.o):(.text+0x274e): undefined reference to `__imp_nghttp2_pack_settings_payload'
ld: libcurl.a(libcurl_la-http2.o):(.text+0x2883): undefined reference to `__imp_nghttp2_session_want_write'
ld: libcurl.a(libcurl_la-http2.o):(.text+0x28ee): undefined reference to `__imp_nghttp2_session_resume_data'
ld: libcurl.a(libcurl_la-http2.o):(.text+0x2909): undefined reference to `__imp_nghttp2_session_want_write'
ld: libcurl.a(libcurl_la-http2.o):(.text+0x2a95): undefined reference to `__imp_nghttp2_session_callbacks_new'
ld: libcurl.a(libcurl_la-http2.o):(.text+0x2aaf): undefined reference to `__imp_nghttp2_session_callbacks_set_send_callback'
ld: libcurl.a(libcurl_la-http2.o):(.text+0x2ac1): undefined reference to `__imp_nghttp2_session_callbacks_set_on_frame_recv_callback'
ld: libcurl.a(libcurl_la-http2.o):(.text+0x2ad3): undefined reference to `__imp_nghttp2_session_callbacks_set_on_data_chunk_recv_callback'
ld: libcurl.a(libcurl_la-http2.o):(.text+0x2ae5): undefined reference to `__imp_nghttp2_session_callbacks_set_on_stream_close_callback'
ld: libcurl.a(libcurl_la-http2.o):(.text+0x2af7): undefined reference to `__imp_nghttp2_session_callbacks_set_on_begin_headers_callback'
ld: libcurl.a(libcurl_la-http2.o):(.text+0x2b09): undefined reference to `__imp_nghttp2_session_callbacks_set_on_header_callback'
ld: libcurl.a(libcurl_la-http2.o):(.text+0x2b1b): undefined reference to `__imp_nghttp2_session_callbacks_set_error_callback'
ld: libcurl.a(libcurl_la-http2.o):(.text+0x2b30): undefined reference to `__imp_nghttp2_session_client_new'
ld: libcurl.a(libcurl_la-http2.o):(.text+0x2b3d): undefined reference to `__imp_nghttp2_session_callbacks_del'
ld: libcurl.a(libcurl_la-http2.o):(.text+0x2cb7): undefined reference to `__imp_nghttp2_submit_settings'
ld: libcurl.a(libcurl_la-http2.o):(.text+0x2cd9): undefined reference to `__imp_nghttp2_session_set_local_window_size'
ld: libcurl.a(libcurl_la-http2.o):(.text+0x2d70): undefined reference to `__imp_nghttp2_session_upgrade'
ld: libcurl.a(libcurl_la-http2.o):(.text+0x2d90): undefined reference to `__imp_nghttp2_session_set_stream_user_data'
ld: libcurl.a(libcurl_la-http2.o):(.text+0x2e12): undefined reference to `__imp_nghttp2_strerror'
ld: libcurl.a(libcurl_la-http2.o):(.text+0x2e3a): undefined reference to `__imp_nghttp2_strerror'
ld: libcurl.a(libcurl_la-http2.o):(.text+0x2e6a): undefined reference to `__imp_nghttp2_strerror'
ld: libcurl.a(libcurl_la-http2.o):(.text+0x2ec8): undefined reference to `__imp_nghttp2_session_set_local_window_size'
ld: libcurl.a(libcurl_la-http2.o):(.text+0x2f04): undefined reference to `__imp_nghttp2_strerror'
```

With "-l:libnghttp2.dll.a" - build passed, but it's not static.

Same with and without #8469 applied (default `-openssl` variant)

--- a/mingw-w64-curl/PKGBUILD
+++ b/mingw-w64-curl/PKGBUILD
@@ -13,7 +13,7 @@ fi
 _realname=curl
 pkgbase=mingw-w64-${_realname}
 pkgname="${MINGW_PACKAGE_PREFIX}-${_realname}${_namesuff}"
-pkgver=7.76.1
+pkgver=7.77.0
 pkgrel=1
 pkgdesc="Command line tool and library for transferring data with URLs. (mingw-w64)"
 arch=('any')
@@ -49,7 +49,7 @@ source=("https://github.com/curl/curl/releases/download/${_realname}-${pkgver//.
         "0001-Make-cURL-relocatable.patch"
         "0002-nghttp2-static.patch"
         "0003-libpsl-static-libs.patch")
-sha256sums=('64bb5288c39f0840c07d077e30d9052e1cbb9fa6c2dc52523824cc859e679145'
+sha256sums=('0f64582c54282f31c0de9f0a1a596b182776bd4df9a4c4a2a41bbeb54f62594b'
             'SKIP'
             '41e7d1f734cae5fbeea8a84ec4af40c29bf7ffbf156ab290abd57df840312ca1'
             '79eec8a337e375d5102fef884030ceacd163a79e5c495e9a974a6b9a11b60c61'
@@ -74,34 +74,31 @@ build() {
   else
     extra_config+=( --disable-debug )
   fi
-  mkdir -p "${srcdir}/build-${CARCH}"
+
   local -a _variant_config
   if [ "${_variant}" = "-winssl" ]; then
-    _variant_config+=("--without-ssl")
-    _variant_config+=("--without-gnutls")
-    _variant_config+=("--with-winssl")
+    _variant_config+=("--with-schannel")
     _variant_config+=('--without-nghttp2')
     _variant_config+=("--without-ca-bundle")
     _variant_config+=("--without-ca-path")
     _variant_config+=("--without-librtmp")
   elif [ "${_variant}" = "-gnutls" ]; then
     _variant_config+=("--with-default-ssl-backend=gnutls")
-    _variant_config+=("--without-ssl")
     _variant_config+=("--with-gnutls")
-    _variant_config+=("--with-winssl")
+    _variant_config+=("--with-schannel")
     _variant_config+=('--without-nghttp2')
     _variant_config+=("--with-ca-bundle=${MINGW_PREFIX}/ssl/certs/ca-bundle.crt")
     _variant_config+=("--with-librtmp")
   elif [ "${_variant}" = "-openssl" ]; then
     _variant_config+=("--with-default-ssl-backend=openssl")
-    _variant_config+=("--without-gnutls")
-    _variant_config+=("--with-ssl")
-    _variant_config+=("--with-winssl")
+    _variant_config+=("--with-openssl")
+    _variant_config+=("--with-schannel")
     _variant_config+=("--with-ca-bundle=${MINGW_PREFIX}/ssl/certs/ca-bundle.crt")
-    _variant_config+=("--with-nghttp2=${MINGW_PREFIX}/")
+    _variant_config+=("--with-nghttp2=${MINGW_PREFIX}")
     _variant_config+=("--without-librtmp")
   fi
-  cd "${srcdir}/build-${CARCH}"
+
+  mkdir -p "${srcdir}/build-static-${MSYSTEM}" && cd "${srcdir}/build-static-${MSYSTEM}"
   ../${_realname}-${pkgver}/configure \
     --prefix=${MINGW_PREFIX} \
     --build=${MINGW_CHOST} \
@@ -109,6 +106,30 @@ build() {
     --target=${MINGW_CHOST} \
     --without-random \
     --enable-static \
+    --disable-shared \
+    --enable-sspi \
+    --enable-ldap \
+    --enable-ldaps \
+    --with-brotli \
+    --with-ldap-lib=wldap32 \
+    --with-libmetalink=${MINGW_PREFIX} \
+    --with-libssh2 \
+    --with-zstd \
+    "${_variant_config[@]}" \
+    "${extra_config[@]}"
+# there's a bug with zsh completion generation script and Windows.
+# curl has to specified with the file extension.
+  sed -i "s|\/curl > \$\@|\/curl\$\{EXEEXT\} > \$\@|" scripts/Makefile
+  make
+
+  mkdir -p "${srcdir}/build-shared-${MSYSTEM}" && cd "${srcdir}/build-shared-${MSYSTEM}"
+  ../${_realname}-${pkgver}/configure \
+    --prefix=${MINGW_PREFIX} \
+    --build=${MINGW_CHOST} \
+    --host=${MINGW_CHOST} \
+    --target=${MINGW_CHOST} \
+    --without-random \
+    --disable-static \
     --enable-shared \
     --enable-sspi \
     --enable-ldap \
@@ -127,7 +148,10 @@ build() {
 }
 
 package() {
-  cd "${srcdir}/build-${CARCH}"
+  cd "${srcdir}/build-static-${MSYSTEM}"
+  make DESTDIR="${pkgdir}" install
+
+  cd "${srcdir}/build-shared-${MSYSTEM}"
   make DESTDIR="${pkgdir}" install
 
   local PREFIX_DEPS=$(cygpath -am ${MINGW_PREFIX})

